src/overlays/gamestates/ovl_file_choose/file_select.h

```diff
@@ -0,0 +1,334 @@
+    /* 28 */ CM_ERASE_ANIM_1,
+    /* 29 */ CM_ERASE_WAIT_FOR_FLASH_SAVE,
+    /* 30 */ CM_ERASE_ANIM_2,
+    /* 31 */ CM_ERASE_ANIM_3,
+    /* 32 */ CM_EXIT_ERASE_TO_MAIN,
+    /* 33 */ CM_UNUSED_31,
+    /* 34 */ CM_ROTATE_TO_NAME_ENTRY,
+    /* 35 */ CM_START_NAME_ENTRY,
+    /* 36 */ CM_NAME_ENTRY,
+    /* 37 */ CM_NAME_ENTRY_WAIT_FOR_FLASH_SAVE,
+    /* 38 */ CM_NAME_ENTRY_TO_MAIN,
+    /* 39 */ CM_MAIN_TO_OPTIONS,
+    /* 40 */ CM_START_OPTIONS,
+    /* 41 */ CM_OPTIONS_MENU,
+    /* 42 */ CM_OPTIONS_WAIT_FOR_FLASH_SAVE,
+    /* 43 */ CM_OPTIONS_TO_MAIN,
+    /* 44 */ CM_UNUSED_DELAY
+} ConfigMode;
+
+typedef enum {
+    /* 0 */ SM_FADE_MAIN_TO_SELECT,
+    /* 1 */ SM_MOVE_FILE_TO_TOP,
+    /* 2 */ SM_FADE_IN_FILE_INFO,
+    /* 3 */ SM_CONFIRM_FILE,
+    /* 4 */ SM_FADE_OUT_FILE_INFO,
+    /* 5 */ SM_MOVE_FILE_TO_SLOT,
+    /* 6 */ SM_FADE_OUT,
+    /* 7 */ SM_LOAD_GAME
+} SelectMode;
+
+typedef enum {
+    /* 0 */ FS_TITLE_SELECT_FILE,   // "Please select a file."
+    /* 1 */ FS_TITLE_OPEN_FILE,     // "Open this file?"
+    /* 2 */ FS_TITLE_COPY_FROM,     // "Copy which file?"
+    /* 3 */ FS_TITLE_COPY_TO,       // "Copy to which file?"
+    /* 4 */ FS_TITLE_COPY_CONFIRM,  // "Are you sure?"
+    /* 5 */ FS_TITLE_COPY_COMPLETE, // "File copied."
+    /* 6 */ FS_TITLE_ERASE_FILE,    // "Erase which file?"
+    /* 7 */ FS_TITLE_ERASE_CONFIRM, // "Are you sure?"
+    /* 8 */ FS_TITLE_ERASE_COMPLETE // "File erased."
+} TitleLabel;
+
+typedef enum {
+    /* -1 */ FS_WARNING_NONE = -1,
+    /*  0 */ FS_WARNING_NO_FILE_COPY,   // "No file to copy."
+    /*  1 */ FS_WARNING_NO_FILE_ERASE,  // "No file to erase."
+    /*  2 */ FS_WARNING_NO_EMPTY_FILES, // "There is no empty file."
+    /*  3 */ FS_WARNING_FILE_EMPTY,     // "This is an empty file."
+    /*  4 */ FS_WARNING_FILE_IN_USE     // "This file is in use."
+} WarningLabel;
+
+typedef enum {
+    /* 0 */ FS_BTN_MAIN_FILE_1,
+    /* 1 */ FS_BTN_MAIN_FILE_2,
+    /* 2 */ FS_BTN_MAIN_FILE_3,
+    /* 3 */ FS_BTN_MAIN_COPY,
+    /* 4 */ FS_BTN_MAIN_ERASE,
+    /* 5 */ FS_BTN_MAIN_OPTIONS
+} MainMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_COPY_FILE_1,
+    /* 1 */ FS_BTN_COPY_FILE_2,
+    /* 2 */ FS_BTN_COPY_FILE_3,
+    /* 3 */ FS_BTN_COPY_QUIT
+} CopyMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_ERASE_FILE_1,
+    /* 1 */ FS_BTN_ERASE_FILE_2,
+    /* 2 */ FS_BTN_ERASE_FILE_3,
+    /* 3 */ FS_BTN_ERASE_QUIT
+} EraseMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_SELECT_FILE_1,
+    /* 1 */ FS_BTN_SELECT_FILE_2,
+    /* 2 */ FS_BTN_SELECT_FILE_3,
+    /* 3 */ FS_BTN_SELECT_YES,
+    /* 4 */ FS_BTN_SELECT_QUIT
+} SelectMenuButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_CONFIRM_YES,
+    /* 1 */ FS_BTN_CONFIRM_QUIT
+} ConfirmButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_BTN_ACTION_COPY,
+    /* 1 */ FS_BTN_ACTION_ERASE
+} ActionButtonIndex;
+
+typedef enum {
+    /* 0 */ FS_SETTING_AUDIO,
+    /* 1 */ FS_SETTING_TARGET
+} SettingIndex;
```

Comment:
z-targetting (n64) or l-targetting (gamecube). i.e. this is where you choose "switch targeting" or "hold targeting" in the menu

Reply:
Oh that kind of settings index, hmm, maybe `_ZTARGET` ? Maybe this one is just me and should be ignored

Reply:
It's an open question whether we should use `ZTargeting` or just `Targeting` because of the ambiguity of GameCube versions turning it into `LTargeting`. I do like `Ztargeting`, but others don't. Currently there's a mix of both used in OoT/MM